TOOLS/ce/src/main/scala/org.opalj.ce/DocumentationComment.scala

```diff
@@ -9,12 +9,7 @@ import org.apache.commons.text.StringEscapeUtils
 /**
  * Container for the comments of a config node.
  */
-class DocumentationComment {
-    val constraints: ListBuffer[String] = ListBuffer[String]()
-    val description: ListBuffer[String] = ListBuffer[String]()
-    var label = ""
-    var brief = ""
-    var datatype = ""
+class DocumentationComment(val label: String, val brief: String, val description: Seq[String], val datatype: String, val constraints: Seq[String]) {
```

Comment:
Should the dataType rather be an enum or case objects extending a (sealed?) trait rather than a String?

Reply:
No, because datatype is a flag that is directly parsed from the "// @type something" comment from the config file.
While ce only implements special handling for subclass and enum types, other types will simply be printed to the documentation as-is. 

The advantage is that types that may exist, but are not explicitly names (such as bool, int, double, etc.) can be used as allowed types too. These types have their own, well known constraints and improve understandability without requiring more code in ce.